### DISCUSSION TITLE
Is it possible to completely remove / disable the default keyboard / text input?

### DISCUSSION THREAD

**Discussion by Evertt** [telegraf/telegraf#1819] [Q&A]
When a user gives my bot the `/settings` command, I make them enter a `scene` I set up for viewing settings and changing them. It's not pretty at all yet, but here's what it looks like so far:

<img width="492" alt="image" src="https://user-images.githubusercontent.com/1267282/226904810-c1f07d07-3bcd-4df7-9d90-3532150c4c88.png">

Since this is a scene, I would like the user to not have any other option available to them except for the options given by the inline keyboard. Because if the user starts writing their own random messages, then those messages will be responded to by handlers outside of the scene. And that makes things messy in my mind.

So I would like to be able to completely disable (or even remove) the text input, which I will only re-enable after they have left the scene.

Is that possible to do?

**Comment by MKRhere** [comment]
> will be responded to by handlers outside of the scene

This is not true, the scene will be able to handle updates as long as the stage is mounted after all other handlers. There's no way to block all input. The best you can do is ignore or reply with an error message.

**Comment by Evertt** [comment]
@MKRhere 

> as long as the stage is mounted after all other handlers

Oh really? Okay that's good to know.

I only today became aware that `bot.command("mycommand", ctx => {})` doesn't work if you place it after `bot.on(message('text'), ctx => {})`. The command handlers have to become _before_ the other text message handlers. Which I found a bit confusing, because I thought it should be obvious that a text message is actually a command if it starts with a `/` and is just a single word, right?

## Edit

By the way, do you happen to have any tips on how to make a settings menu that has a pretty UI and good UX?

**Reply by MKRhere** [reply]
> obvious that a text message is actually a command if it starts with a `/`

That is precisely the case. The inverse of this statement is that "commands are actually text messages with a command entity". So command messages are matched as text if that is mounted first.
